### DISCUSSION TITLE
Application In Object Detection

### DISCUSSION THREAD

**Discussion by Zengyf-CVer** [frgfm/torch-cam#85] [Ideas]
@frgfm 
Hello, I am using your CAM program for the first time. I would like to ask, is this program suitable for object detection?
For example, I now use yolov5 to train a model, and then want to use your CAM program to export the heatmap of the detected image. Is this feasible? Although I slightly know that CAM is aimed at classification problems, the detection also includes classification.

**Comment by frgfm** [comment]
Hi @Zengyf-CVer :wave:

So by default, it's not possible. Simply put, this repo leverages module hooks to easily manipulate what it needs to compute the CAMs. Most object detection models have various return & input types, which makes it hard to handle.

So having something to work for all object detection models is really tricky. I'll try to think about how to modify them. However, by modifying parts of the code, it's possible to get the CAM to work for your own architecture!

**Comment by frgfm** [comment]
Hi @Zengyf-CVer,
Moving this to a discussion as it sounds more appropriate for this topic :+1:
Let me know if you have some feedback on my previous message :point_up:
